src/OpenTelemetry.Sampler.AWS/README.md

```diff
@@ -15,20 +15,34 @@ dotnet add package OpenTelemetry.Sampler.AWS
 
 You can configure the `AWSXRayRemoteSampler` as per the following example.
 Note that you will need to configure your [OpenTelemetry Collector for
-X-Ray remote sampling](https://aws-otel.github.io/docs/getting-started/remote-sampling)
+X-Ray remote sampling](https://aws-otel.github.io/docs/getting-started/remote-sampling).
+This example also sets up the Console Exporter,
+which requires adding the package [`OpenTelemetry.Exporter.Console`](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.Console/README.md)
+to the application.
 
 ```csharp
 using OpenTelemetry;
+using OpenTelemetry.Contrib.Extensions.AWSXRay.Resources;
 using OpenTelemetry.Contrib.Extensions.AWSXRay.Trace;
+using OpenTelemetry.Sampler.AWS;
 using OpenTelemetry.Trace;
 
-var tracerProvider = Sdk.CreateTracerProviderBuilder()
-                        // other configurations
-                        .SetSampler(AWSXRayRemoteSampler.Builder()
-                                                        .SetPollingInterval(TimeSpan.FromSeconds(10))
-                                                        .SetEndpoint("http://localhost:2000")
-                                                        .Build())
-                        .Build();
+var serviceName = "MyServiceName";
+
+var resourceBuilder = ResourceBuilder
+    .CreateDefault()
+    .AddService(serviceName: serviceName)
+    .AddDetector(new AWSEC2ResourceDetector());
+
+using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+    .AddSource(serviceName)
+    .SetResourceBuilder(resourceBuilder)
+    .AddConsoleExporter()
+    .SetSampler(AWSXRayRemoteSampler.Builder(resourceBuilder.Build()) // you must provide a resource
```

Comment:
Just a suggestion:

This setup looks quite verbose. I think you could extract this into an extension method. Something like: `AddAWSXRayRemoteSampler(this TracerProviderBuilder, Action<AWSXRaySamplerOptions> configure)`.

That way you would only be exposing the required options and you could reduce the publicAPI surface by marking these methods as internal: `AWSXRayRemoteSampler.Builder`, `SetPollingInterval`, `SetEndpoint`, `Build`

Reply:
I had earlier of thought of doing the extension method similar to `AddXRayTraceId()` we have for the custom trace id generator, but I think that is counterintuitive for adding a Sampler. Since the `AWSXRayRemoteSampler` is just another implementation of `Sampler`, it should follow the standard procedure of adding a sampler via the `SetSampler` method.

Regarding the setting of the sampler being verbose, I agree. The customers of OTel [Java](https://aws-otel.github.io/docs/getting-started/java-sdk/trace-manual-instr#using-x-ray-remote-sampling) and [GoLang](https://aws-otel.github.io/docs/getting-started/go-sdk/trace-manual-instr#sampling-using-aws-x-ray-remote-sampler) are already configuring the remote sampler in similar way, so I would like to keep the same behavior here as well.